### PR TITLE
feat: add support for Gatsby v4

### PR DESCRIPTION
**Pull request by jpedroschmitz**
This PR adds support for Gatsby v4.

Looking into the migration guide, we don't have any relevant updates to do in the package.

We also need to publish it under the `next` tag on npm.

### CODE REVIEW

**Comment by jpedroschmitz** [comment]
The build is failing because of the Node.js version.
Gatsby v4 requires Node.js >=14.15. The project still uses version 12 - I couldn't find a place to change it.

File: gatsby-source-graphcms/README.md [hunk 1]
```diff
@@ -22,10 +22,18 @@
     <a href="https://gatsby-source-graphcms.vercel.app">Demo</a> • <a href="https://github.com/GraphCMS/gatsby-starter-graphcms-blog">gatsby-starter-graphcms-blog</a> • <a href="https://slack.graphcms.com">Join us on Slack</a>  • <a href="https://app.graphcms.com">Login to GraphCMS</a> • <a href="https://twitter.com/GraphCMS">@GraphCMS</a>
 </p>
 
+<br />
```
Comment (notrab) [inline gatsby-source-graphcms/README.md:25]: Is this needed?

**Review by notrab — APPROVED** [review]

File: gatsby-source-graphcms/README.md (hunk 1 above)
Comment (jpedroschmitz) [reply gatsby-source-graphcms/README.md:25]: It gives a little margin from the badges. It looks better this way IMO.

**Review by notrab — APPROVED** [review]

**Comment by jpedroschmitz** [comment]
It's building correctly! The bot hasn't updated the status.

**Comment by notrab** [comment]
🤖

**Pull request merged**